lib/iris/cube.py

```diff
@@ -1638,18 +1633,20 @@ def dtype(self):
 
     @dtype.setter
     def dtype(self, dtype):
```

Comment:
This setter method all seems very unstable... hopefully we will be able to reduce the scope of this (or remove it entirely!) with upcoming changes.